File: app/src/main/java/com/settlex/android/ui/dashboard/model/UserUiModel.java
package com.settlex.android.ui.dashboard.model;

import com.google.firebase.Timestamp;

public class UserUiModel {
    private final String uid, email, firstName, lastName, phone, paymentId, photoUrl;
    boolean hasPin;
    private final long balance, commissionBalance, referralBalance;
    private final Timestamp createdAt;

    public UserUiModel(String uid, String email, String firstName, String lastName, Timestamp createdAt, String phone, String paymentId, String photoUrl, boolean hasPin, long balance, long commissionBalance, long referralBalance) {
        this.uid = uid;
        this.email = email;
        this.firstName = firstName;
        this.lastName = lastName;
        this.createdAt = createdAt;
        this.paymentId = paymentId;
        this.photoUrl = photoUrl;
        this.hasPin = hasPin;
        this.phone = phone;
        this.balance = balance;
        this.commissionBalance = commissionBalance;
        this.referralBalance = referralBalance;
    }

    // GETTERS
    public String getUid() {
        return uid;
    }

    public String getEmail() {
        return email;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public String getPaymentId() {
        return paymentId;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public boolean hasPin() {
        return hasPin;
    }

    public String getPhone() {
        return phone;
    }

    public long getBalance() {
        return balance;
    }

    public long getCommissionBalance() {
        return commissionBalance;
    }

    public long getReferralBalance() {
        return referralBalance;
    }

    public String getFullName() {
        return getFirstName() + " " + getLastName();
    }
}

File: app/src/main/java/com/settlex/android/presentation/account/model/UserUiModel.kt
package com.settlex.android.presentation.account.model

import com.google.firebase.Timestamp

data class UserUiModel(
    @JvmField val uid: String,
    @JvmField val email: String,
    @JvmField val firstName: String,
    @JvmField val lastName: String,
    @JvmField val createdAt: Timestamp,
    @JvmField val phone: String,
    @JvmField val paymentId: String?,
    @JvmField val photoUrl: String?,
    @JvmField val hasPin: Boolean,
    @JvmField val balance: Long,
    @JvmField val commissionBalance: Long,
    @JvmField val referralBalance: Long
) {

    val fullName: String get() = "$firstName $lastName"

    fun hasPin(): Boolean {
        return hasPin
    }
}